app/models/manageiq/providers/vmware/infra_manager/provision/cloning.rb

```diff
@@ -2,14 +2,19 @@ module ManageIQ::Providers::Vmware::InfraManager::Provision::Cloning
   def do_clone_task_check(clone_task_mor)
     source.with_provider_connection do |vim|
       begin
-        state, val = vim.pollTask(clone_task_mor, "VMClone")
-        case state
+        task_props = ["info.state", "info.error", "info.result", "info.progress", "info.completeTime"]
+        task = vim.getMoProp(clone_task_mor, task_props)
+
+        case task&.info&.state
```

Comment:
Overall this looks good.  Since we are referencing `task&.info` in 5 places we should set that to a variable to avoid the lookup each time: `task_info = task&.info`

Reply:
@gmcculloug done